src/mon/OSDMonitor.cc

```diff
@@ -2108,7 +2109,9 @@ bool OSDMonitor::prepare_boot(MonOpRequestRef op)
   if (osdmap.is_up(from)) {
     dout(7) << "prepare_boot was up, first marking down " << osdmap.get_inst(from) << dendl;
     // preprocess should have caught these;  if not, assert.
-    assert(osdmap.get_inst(from) != m->get_orig_source_inst());
+    bool addr_changed = (osdmap.get_inst(from) != m->get_orig_source_inst() ||
```

Comment:
could also put ``osdmap.get_cluster_addr(from) != m->cluster_addr)`` on a new line. like
```c
assert(osdmap.get_inst(from) != m->get_orig_source_inst() ||
       osdmap.get_cluster_addr(from) != m->cluster_addr);
```

 as ideally, `assert()` is a no-op in a NDEBUG build, and compiler or static analyzer would complain at seeing a variable not being used anywhere.